lib/NodeUtils.js

```diff
@@ -98,6 +98,49 @@ function attrname(a) {
   return a.name;
 }
 
+/**
+ * Escapes matching closing tag in a raw text.
+ *
+ * For example, given `<style>#text(</style><script></script>)</style>`,
+ * the parent tag would by "style" and the raw text is
+ * "</style><script></script>". If we come across a matching closing tag
+ * (in out case `</style>`) - replace `<` with `&lt;` to avoid unexpected
+ * and unsafe behavior after de-serialization.
+ */
+function escapeMatchingClosingTag(rawText, parentTag) {
```

Comment:
Nit: from a performance point of view would it be more optimal to do an initial scan to check for either a closing tag `</` or the closing pattern `'</' + parentTag.toLowerCase() + '>';`?